ts/Data/GoogleDataStore.ts

```diff
@@ -0,0 +1,227 @@
+            gc = cell.gs$cell.col - 1; // columns start at 1
+
+            // If both row and col falls inside start and end set the
+            // transposed cell value in the newly created columns
+            if (gc >= startColumn && gc <= endColumn &&
+                gr >= startRow && gr <= endRow) {
+
+                cellInner = cell.gs$cell || cell.content;
+
+                val = null;
+
+                if (cellInner.numericValue) {
+                    if (cellInner.$t.indexOf('/') >= 0 ||
+                        cellInner.$t.indexOf('-') >= 0) {
+                        // This is a date - for future reference.
+                        val = cellInner.$t;
+                    } else if (cellInner.$t.indexOf('%') > 0) {
+                        // Percentage
+                        val = parseFloat(cellInner.numericValue) * 100;
+                    } else {
+                        val = parseFloat(cellInner.numericValue);
+                    }
+                } else if (cellInner.$t && cellInner.$t.length) {
+                    val = cellInner.$t;
+                }
+
+                columns[gc - startColumn][gr - startRow] = val;
+            }
+        }
+
+        // Insert null for empty spreadsheet cells (#5298)
+        columns.forEach(function (
+            column: Array<Highcharts.DataValueType>
+        ): void {
+            for (i = 0; i < column.length; i++) {
+                if (typeof column[i] === 'undefined') {
+                    column[i] = null as any;
+                }
+            }
+        });
+
+        /* *
+         * TODO:
+         * add data to dataTable
+         * ...
+         *
+         * */
+    }
+
+    private fetchSheet(): (boolean) {
+        const parseSheet = this.parseSheet;
+        const fetchSheet = this.fetchSheet;
+
+        const enablePolling = this.enablePolling;
+        const dataRefreshRate = this.dataRefreshRate;
+
+        const url = [
+            'https://spreadsheets.google.com/feeds/cells',
+            this.googleSpreadsheetKey,
+            this.worksheet,
+            'public/values?alt=json'
+        ].join('/');
+
+        ajax({
+            url: url,
+            dataType: 'json',
+            success: function (json: Highcharts.JSONType): void {
+
+                parseSheet(json);
+
+                // Polling
+                if (enablePolling) {
+                    setTimeout(
+                        function (): void {
+                            fetchSheet();
+                        },
+                        dataRefreshRate * 1000
+                    );
+                }
+            },
+            error: function (
+                xhr: XMLHttpRequest,
+                text: (string|Error)
+            ): void {
+                /* *
+                 * TODO:
+                 * catch error
+                 * ...
+                 *
+                 * */
+                // console.log(text);
+            }
```

Comment:
We need something here, to inform anyone when the async load is finished and the dataTable is available.

* One way would be a callback parameter taken by `load` and passed through to `fetchSheet`. The `callback` function should receive either a `DataTable` instance or an `Error` instance.

* Alternatively we can fire events `load`/`afterLoad`, `parse`/`afterParse` and `fail`, which would be similar to `DataTable` events and maybe more flexible. Any opinions?

Reply:
Maybe we should consider to use `await / async` ?

From two suggestions, I am closer to the solutuon based on events. It sounds better because of consistency (of both modules).

Any other comments here?

Reply:
We unfortunately can not, because of our browser target ES5. `await` / `async` and `Promise` are not part of it.

Reply:
Ahh indeed, I always do not remember about that. 

So well, events sounds better than callbacks, but maybe lets wait for other comments?

Reply:
In my opinion, events would be a better choice.

Reply:
So, we will try to implement events for now if no other opinions.